com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/util/ApplicationURI.java

```diff
@@ -0,0 +1,258 @@
+package com.sap.cloud.lm.sl.cf.core.util;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.sap.cloud.lm.sl.cf.core.model.SupportedParameters;
+import com.sap.cloud.lm.sl.cf.core.validators.parameters.PortValidator;
+
+public class ApplicationURI {
+
+    private static final PortValidator PORT_VALIDATOR = new PortValidator();
+
+    public static final String DEFAULT_SCHEME_SEPARATOR = "://";
+    public static final char DEFAULT_PATH_SEPARATOR = '/';
+    public static final char DEFAULT_HOST_DOMAIN_SEPARATOR = '.';
+    public static final char DEFAULT_PORT_SEPARATOR = ':';
+
+    public static final String UNPARSABLE_PORT_PARAMETER = "unparsable-port";
+    public static final String SCHEME_PARAMETER = "scheme";
+
+    public static final int STANDARD_HTTP_PORT = 80;
+    public static final int STANDARD_HTTPS_PORT = 443;
+
+    public static final String TCP_PROTOCOL = "tcp";
+    public static final String TCPS_PROTOCOL = "tcps";
+    public static final String HTTP_PROTOCOL = "http";
+    public static final String HTTPS_PROTOCOL = "https";
+
+    private String uri;
+    private boolean isHostBased;
+
+    private Map<String, Object> uriParts;
+
+    public static Integer getPortFromURI(String uri) {
+        ApplicationURI parsedURI = new ApplicationURI(uri);
+
+        return parsedURI.getParsedPort();
+    }
+
+    public static String getDomainFromURI(String uri) {
+        ApplicationURI parsedURI = new ApplicationURI(uri);
+
+        return parsedURI.getDomain();
+    }
+
+    public ApplicationURI(String initial) {
+        uriParts = new HashMap<>();
+        uri = stripScheme(initial);
+
+        int portIndex = uri.lastIndexOf(DEFAULT_PORT_SEPARATOR);
+        if (portIndex == -1) {
```

Comment:
Do `isHostBased = portIndex == -1` and then you wouldn't need to initialize it in the `initHostBased` and `initPortBased` methods.